File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Events/OnUseCardAfterMessage.ts
import { IMessage } from "../../../../Core/ObserverMessageKit";
import AbstractCard from "../Cards/AbstractCard";
import AbstractCreature from "../Unit/AbstractCreature";

export class OnUseCardAfterMessage  implements IMessage{
    public static readonly Id = "OnUseCardAfterMessage";
    constructor(public readonly Source:AbstractCreature,public Target:AbstractCreature,public readonly Card:AbstractCard) {
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Event/Ids/OnUseCardAfterMessage.cs
using ModTheSpire.Card;
using ModTheSpire.Unit;

namespace ModTheSpire.Event.Ids
{
    ///使用卡牌前
    public class OnUseCardAfterMessage : IEventHandle
    {
        public AbstractCreature Source;
        public AbstractCreature Target;      
        public AbstractCard Card;

        public OnUseCardAfterMessage(AbstractCreature source, AbstractCreature target, AbstractCard card)
        {
            Source = source;
            Target = target;
            Card = card;
        }
    }
}